autoload/neomake/makers/ft/rust.vim

```diff
@@ -1,19 +1,24 @@
 " vim: ts=4 sw=4 et
 
-function! neomake#makers#ft#rust#EnabledMakers()
+function! neomake#makers#ft#rust#EnabledMakers() abort
     return ['rustc']
 endfunction
 
-function! neomake#makers#ft#rust#rustc()
+function! neomake#makers#ft#rust#rustc() abort
     return {
-        \ 'args': ['-Z', 'parse-only'],
         \ 'errorformat':
-            \ '%-G%f:%s:,' .
-            \ '%f:%l:%c: %trror: %m,' .
-            \ '%f:%l:%c: %tarning: %m,' .
-            \ '%f:%l:%c: %m,'.
-            \ '%f:%l: %trror: %m,'.
-            \ '%f:%l: %tarning: %m,'.
-            \ '%f:%l: %m',
+            \ '%-Gerror: aborting due to previous error,'.
+            \ '%-Gerror: aborting due to %\\d%\\+ previous errors,'.
+            \ '%-Gerror: Could not compile `%s`.,'.
+            \ '%Eerror[E%n]: %m,'.
+            \ '%Eerror: %m,'.
+            \ '%Wwarning: %m,'.
+            \ '%Inote: %m,'.
+            \ '%-Z\ %#-->\ %f:%l:%c,'.
+            \ '%G\ %#\= %*[^:]: %m,'.
+            \ '%G\ %#|\ %#%\\^%\\+ %m,'.
+            \ '%I%>help:\ %#%m,'.
+            \ '%Z\ %#%m,'.
+            \ '%-G%s,',
```

Comment:
Not sure about the trailing comma here (in the errorformat), better remove it if it's not there on purpose.